indra/tools/reading/readers.py

```diff
@@ -0,0 +1,546 @@
+                        if output_file is not None:
+                            output_file_list.append(output_file)
+                        if log:
+                            outbuf.write('Log for producing %s.\n'
+                                         % output_file)
+                            buff.seek(0)
+                            outbuf.write(buff.read() + '\n')
+                            outbuf.flush()
+            finally:
+                if log:
+                    outbuf.close()
+                    if verbose:
+                        logger.info("Sparser logs may be found at %s." %
+                                    log_name)
+            ret = self.get_output(output_file_list)
+        return ret
+
+
+def get_readers():
+    """Get all children of the Reader objcet."""
+    try:
+        children = Reader.__subclasses_()
+    except AttributeError:
+        module = sys.modules[__name__]
+        children = [cls for cls_name, cls in module.__dict__.items()
+                    if isinstance(cls, type) and issubclass(cls, Reader)
+                    and cls_name != 'Reader']
+    return children
+
+
+class ReadingData(object):
+    """Object to contain the data produced by a reading.
+
+    This is primarily designed for use with the database.
+
+    Init Parameters
+    ---------------
+    tcid : int or str
+        An identifier of the text content that produced the reading. Must
+        be an int for use with the database.
+    reader : str
+        The name of the reader, consistent with it's `name` attribute, for
+        example: 'REACH'
+    reader_version : str
+        A string identifying the version of the underlying nlp reader.
+    content_format : str
+        The format of the content. Options are in indra.db.formats.
+    content : str
+        The content of the reading result. A string in the format given by
+        `content_format`.
+    reading_id : int or None
+        Optional. The id corresponding to the Readings entry in the db.
+    """
+
+    def __init__(self, tcid, reader, reader_version, content_format, content,
+                 reading_id=None):
+        self.reading_id = reading_id
+        self.tcid = tcid
+        self.reader = reader
+        self.reader_version = reader_version
+        self.format = content_format
+        self.content = content
+        return
+
+    @classmethod
+    def get_cols(self):
+        """Get the columns for the tuple returned by `make_tuple`."""
+        return ('text_content_id', 'reader', 'reader_version', 'format',
+                'bytes')
+
+    def get_statements(self):
+        """General method to create statements."""
+        if self.reader == ReachReader.name:
+            if self.format == formats.JSON:
+                # Process the reach json into statements.
+                json_str = json.dumps(self.content)
+                stmts = reach.process_json_str(json_str).statements
```

Comment:
It would be better to break this up into two lines, because `reach.process_json_str` returns None in some cases. We should check for None  first, and if the result is not None, access the statements attribute.

Reply:
done